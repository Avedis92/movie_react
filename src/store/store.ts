import { configureStore } from '@reduxjs/toolkit';
import genreReducer from './slices/genre';

export const store = configureStore({
  reducer: {
    genre: genreReducer,
  },
});

export type RootState=ReturnType<typeof store.getState>;
export type MovieDispatch=typeof store.dispatch;
