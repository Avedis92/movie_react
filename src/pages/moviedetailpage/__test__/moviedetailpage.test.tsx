import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '../../../store/store';
import MovieDetailPage from '../moviedetailpage';

jest.mock('react-router-dom', () => ({
  useLocation: jest.fn().mockImplementation(() => ({
    pathname: `/moviedetailpage?movieId=${550}`,
  })),
  useHistory: jest.fn().mockImplementation(() => ({
    pathname: '/',
  })),
}));

test('Make a snapshot test', () => {
  const {asFragment} = render(
    <Provider store={store}>
      <MovieDetailPage />
    </Provider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
