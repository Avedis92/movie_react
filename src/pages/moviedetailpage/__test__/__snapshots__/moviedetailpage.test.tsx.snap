// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make a snapshot test 1`] = `
<div
  className="pageContainer"
>
  <div
    className="searchHeader"
  >
    <h1
      onClick={[Function]}
    >
      Movies
    </h1>
    <input
      list="movieTitle"
      name="movieSearch"
      onKeyPress={[Function]}
      placeholder="Search"
      type="text"
    />
  </div>
  <div
    className="loader"
  >
    <span />
    <span />
    <span />
  </div>
  <div
    className="movieListContainer"
  >
    <div
      className="loader"
    >
      <span />
      <span />
      <span />
    </div>
    <button
      className="seeMore"
      onClick={[Function]}
      type="button"
    >
      See More
    </button>
  </div>
</div>
`;
